### PR TITLE
GH#24286: fix: guard empty table separator cells

### LINKED ISSUE #24286: Review followup: PR #24282 — GH#24271: fix: tighten report table separator detection
## Unaddressed review bot suggestions

PR #24282 was merged with unaddressed review bot feedback. Each comment
below includes its file path, line number, a direct link to the inline
review comment, and a `diff` fence with the code context the bot was
flagging. Resolved and outdated threads are filtered out via GitHub's
GraphQL review-thread state. Read the relevant lines, decide whether
the suggestion is correct, and either apply the fix or close this issue
with a wontfix rationale.

**Source PR:** https://github.com/marcusquinn/aidevops/pull/24282

---

### You are the triager (worker-is-triager rule)

This issue is **auto-created from review bot output** and dispatched
directly to you. Review bots can be wrong: hallucinated line refs, false
premises about codebase structure, template-driven sweeps without
measurements (see GH#17832-17835 for prior art and `AGENTS.md`
"AI-Generated Issue Quality"). **Do not assume the bot is correct.** Verify before acting.

You must end in exactly one of three outcomes — no fourth "hand it back
to the human" path exists. Humans approve decisions; they do not re-do
analysis.

#### Outcome A — Premise falsified → close the issue

1. Read the cited `file:line` (listed under *Files to modify* below).
2. If the bot's claim is factually wrong (file doesn't exist at that
   line, function doesn't behave as described, "auto-generated" section
   isn't actually auto-generated, etc.), **close the issue** with a
   comment in this shape:

   > **Premise falsified.** \<what the bot claimed\>. \<what the code
   > actually shows, with a `file:line` citation or one-line quote\>.
   > Not acting.

   No PR. No further dispatch. The closing comment trains the next
   session reading this thread and the noise filter.

#### Outcome B — Premise correct + fix is obvious → implement and PR

1. Verify the bot's premise as above.
2. Read the Worker Guidance section below, open a worktree, implement.
3. Open a PR with `Resolves #<this-issue-number>` in the body
   (use THIS issue's number, not the source PR's) so merge auto-closes it.
4. Follow the normal Lifecycle Gate (brief, tests, review-bot-gate,
   merge, postflight).

#### Outcome C — Premise correct but approach is a genuine judgment call

Only use this path if you reach it after Outcomes A and B don't apply:
the bot's finding is real, but the fix requires a decision that is
architectural, policy, breaking-change, or otherwise genuinely outside
what you can resolve autonomously. In that case, post a **decision
comment** with exactly these fields:

- **Premise check:** one line, confirming the finding is real.
- **Analysis:** 2-4 bullets on the trade-offs.
- **Recommended path:** the option you would take if the decision were
  yours, with rationale.
- **Specific question:** the single decision the human needs to make
  (yes/no or pick-one, not open-ended).

Then apply `needs-maintainer-review` and stop. The human wakes up to a
ready-to-approve recommendation, not a blank task.

> **Ambiguity about scope or style is not Outcome C.** Per
> `AGENTS.md` "Reasoning responsibility", the model does the
> thinking and delivers a recommendation. Only escalate what is genuinely
> a maintainer-only decision.

### Worker Guidance

**Files to modify:**

- `.agents/scripts/report_render_blocks.py:69`
- `tests/test_report_render_blocks.py:113`

**Implementation steps (Outcome B path):**

1. Read the `diff` block under each inline comment below — it shows the
   exact code the bot was flagging. Open the file only if you need
   surrounding context beyond what the diff tail shows.
2. Read the bot's full comment below the diff — it contains the rationale
   and any suggested change.
3. Verify the premise before implementing (see Outcome A). If the premise
   is wrong, switch to Outcome A instead of burning iterations trying to
   satisfy a wrong suggestion.
4. If multiple comments target the same file, group your edits into one
   logical commit.
5. Run `shellcheck` / `markdownlint-cli2` / project tests as appropriate.

**Verification:**

- Open the new PR with `Resolves #<this-issue>` so this followup is auto-closed on merge.
- If the bot's suggestion was incorrect, close this issue with a Outcome A comment — do not open a no-op PR.

#### gemini-code-assist on `.agents/scripts/report_render_blocks.py:69`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/24282#discussion_r3321885942)

```diff
@@ -67,6 +63,12 @@ def handle_table(line: str, body: list[str], states: dict[str, object]) -> bool:
     return True
 
 
+def is_markdown_table_separator_row(cells: list[str]) -> bool:
+    """Return True only for a real Markdown table header separator row."""
+    separator_cells = [html.unescape(cell) for cell in cells]
+    return all(re.fullmatch(r":?-{3,}:?", cell) for cell in separator_cells)
```

> medium
> 
> If `cells` is empty, `all()` will return `True`, incorrectly identifying an empty list of cells as a valid table separator row. We should add a guard to return `False` if `cells` is empty.
> 
> ```suggestion
> def is_markdown_table_separator_row(cells: list[str]) -> bool:
>     """Return True only for a real Markdown table header separator row."""
>     if not cells:
>         return False
>     separator_cells = [html.unescape(cell) for cell in cells]
>     return all(re.fullmatch(r":?-{3,}:?", cell) for cell in separator_cells)
> ```

#### gemini-code-assist on `tests/test_report_render_blocks.py:113`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/24282#discussion_r3321885947)

```diff
 
+def test_table_separator_requires_three_dashes_per_cell() -> None:
+    assert_equal(
+        is_markdown_table_separator_row([":---", "---:", ":---:"]),
+        True,
+        "standard alignment separator cells are recognised",
+    )
+    assert_equal(
+        is_markdown_table_separator_row([":-", "-:"]),
+        False,
+        "short dash cells with alignment markers are data, not separators",
+    )
```

> medium
> 
> Add a test case to verify that an empty list of cells is correctly identified as not being a table separator row.
> 
> ```suggestion
> def test_table_separator_requires_three_dashes_per_cell() -> None:
>     assert_equal(
>         is_markdown_table_separator_row([":---", "---:", ":---:"]),
>         True,
>         "standard alignment separator cells are recognised",
>     )
>     assert_equal(
>         is_markdown_table_separator_row([":-", "-:"]),
>         False,
>         "short dash cells with alignment markers are data, not separators",
>     )
>     assert_equal(
>         is_markdown_table_separator_row([]),
>         False,
>         "an empty list of cells is not a separator row",
>     )
> ```

#### gemini-code-assist[bot] review summary

[View review](https://github.com/marcusquinn/aidevops/pull/24282#pullrequestreview-4386285149)

> ## Code Review
> 
> This pull request refactors the markdown table separator row detection logic into a dedicated helper function `is_markdown_table_separator_row` and adds corresponding unit tests. The feedback points out a potential bug where an empty list of cells would be incorrectly identified as a valid separator row because `all()` on an empty iterable returns `True`. It is recommended to add a guard clause for empty lists and include a corresponding test case.

---
[aidevops.sh](https://aidevops.sh) v3.20.5 automated scan.

### PR DESCRIPTION
**Pull request by alex-solovyev**
## Summary

Added an explicit empty-cell guard to Markdown table separator detection and covered it with a regression test.

## Files Changed

.agents/scripts/report_render_blocks.py,tests/test_report_render_blocks.py

## Runtime Testing

- **Risk level:** Low (agent prompts / infrastructure scripts)
- **Verification:** python3 tests/test_report_render_blocks.py

Resolves #24286

---
[aidevops.sh](https://aidevops.sh) v3.20.5 plugin for [OpenCode](https://opencode.ai) v1.15.12 with gpt-5.5 spent 1m and 31,114 tokens on this as a headless worker.

### CODE REVIEW

**Comment by alex-solovyev** [comment]
## Completion Summary

- **What**: Added an explicit empty-cell guard to Markdown table separator detection and covered it with a regression test.
- **Issue**: #24286
- **Files changed**: .agents/scripts/report_render_blocks.py,tests/test_report_render_blocks.py
- **Testing**: python3 tests/test_report_render_blocks.py
- **Key decisions**: none

---
[aidevops.sh](https://aidevops.sh) v3.20.5 plugin for [OpenCode](https://opencode.ai) v1.15.12 with gpt-5.5 spent 1m and 31,114 tokens on this as a headless worker.

**Review by vladimirdulov — APPROVED** [review]
Auto-approved by pulse runner @vladimirdulov — author @alex-solovyev confirmed collaborator, pre-merge gates passed.

**Review by marcusquinn — APPROVED** [review]
Auto-approved by pulse runner @marcusquinn — author @alex-solovyev confirmed collaborator, pre-merge gates passed.
